src/components/Chart/utils/calcMetricPercentage.js

```diff
@@ -14,10 +14,13 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 // =============================================================================
-const createPeriods = (availableDataLength) => [
-  { value: 60, label: "5 years" },
-  { value: 12, label: "1 year" },
-  { value: availableDataLength, label: "All Time" },
-];
+function calcMetricPercentage(data) {
```

Comment:
Should we validate against empty arrays of arrays with length=1 in this function, or is it impossible for such arrays to reach this function as currently constructed? If the latter, we should at least document that in the docstring.

Reply:
It is impossible now that an empty array reaches this function.
An exception added for length=1 array.
Added tests and exceptions for both cases.